### DISCUSSION TITLE
Add multiple objects

### DISCUSSION THREAD

**Discussion by yuichiro-student** [google/brax#493] [Q&A] [answered]
Hi, I am using brax as a physics engine. Specifically, I navigate the agents to the goal while avoiding obstacles. What I would like to ask you here is how I can add markers for the target. My idea is to add a sphere.

By google search, I found some examples of adding objects, but they seem to be old and don't support the current version of brax. For instance, when I run one of the code, it says
 `*** AttributeError: module 'brax.envs.ant' has no attribute '_SYSTEM_CONFIG'`.
Would anyone give me hints for adding markers?
Thank you.
example:
https://colab.research.google.com/gist/DavidSlayback/562f800d145a1fa8fbe4c82c27bb9c3f/ant_gather.ipynb#scrollTo=iFRh-5pChDpB

added
`_SYSTEM_CONFIG` is used only in `v1` of the code. Would you tell me what is the corresponding variable and function to add bodies to it?

**Comment by erikfrey** [comment, marked as answer]
Hello,

Here's an [example of adding a target to a an environment](https://github.com/google/brax/blob/main/brax/envs/assets/reacher.xml#L42) used for Reacher.  The XML specifies a sphere marker for a body that does not move.  Setting `contype` and `conaffinity` to 0 assures that nothing can interact with the target, but it can still be moved in code, see this line in the [reacher env](https://github.com/google/brax/blob/main/brax/envs/reacher.py#L185) for an example.

**Reply by yuichiro-student** [reply]
Thank you, but we don't have a way to add objects in scripts. Is my understanding correct?

**Reply by erikfrey** [reply]
You can use [PyMJCF](https://github.com/google-deepmind/dm_control/blob/main/dm_control/mjcf/README.md) to programmatically change MuJoCo scenes.
